src/gc.c

```diff
@@ -396,9 +405,17 @@ static void run_finalizers(jl_task_t *ct)
     }
     jl_atomic_store_relaxed(&jl_gc_have_pending_finalizers, 0);
     arraylist_new(&to_finalize, 0);
+
+    uint64_t save_rngState[4];
```

Comment:
Should this also apply to `jl_finalize_th`?

Reply:
I guess we should to be consistent, but it's not essential since it would appear explicitly in user code.

Reply:
Well, we should define what the semantics are regardless with a view towards #44056. If regular finalize doesn't do this, we do need a version that does for that purpose.

Reply:
We discussed this and decided that since explicit `finalize` is synchronous, it can just use the current task RNG as before. Since eager finalizer call insertion and finalizer inlining require `:notaskstate` (IIUC) this should be fine right?